src/parser/TripleComponent.h

```diff
@@ -33,7 +33,7 @@ class TripleComponent {
     // TODO: Without this, the code seg faults when creating a `VALUES` caluse
     // containing an `UNDEF` value. Report this to the `absl` people.
     template <typename H>
-    friend H AbslHashValue(H h, const UNDEF& undef) {
+    friend H AbslHashValue(H h, [[maybe_unused]] const UNDEF& undef) {
       return H::combine(std::move(h), 42);
     }
```

Comment:
I played with this a little bit, and the problem is not `absl`, but that `UNDEF` is (deliberately) implicitly convertible
to `TripleComponent`, and thus we will get an infinite loop basically because then the hash function of `TripleComponent` is used to hash `UNDEF`. We can disallow this by making changing

`friend H AbslHashValue(H, const TripleComponent& tc)` (further down) to

```
friend H AbslHashValue(H h, const std::same_as<TripleComponent> auto& tc) {...}
``` 
This function now only accepts `TripleComponent` and not `things that are implicitly convertible to TripleComponent`. (One of the many reasons, why implicit conversion are always a bit fishy.